server/conn.go

```diff
@@ -1706,23 +1706,20 @@ func (cc *clientConn) handleQuery(ctx context.Context, sql string) (err error) {
 		}
 		retryable, err = cc.handleStmt(ctx, stmt, parserWarns, i == len(stmts)-1)
 		if err != nil {
+			if !retryable || !errors.ErrorEqual(err, storeerr.ErrTiFlashServerTimeout) {
+				break
+			}
 			_, allowTiFlashFallback := cc.ctx.GetSessionVars().AllowFallbackToTiKV[kv.TiFlash]
-			if allowTiFlashFallback && errors.ErrorEqual(err, storeerr.ErrTiFlashServerTimeout) && retryable {
+			if allowTiFlashFallback {
```

Comment:
The code logic changes after the modification. If `allowTiFlashFallback` is false, we still need to jump out of the loop.

Reply:
Thanks, fixed in the next commit. The main idea is using guard pattern to prevent deep nest of if branches.